### DISCUSSION TITLE
SharePointDocumentLocations and SharePointSiteCollections

### DISCUSSION THREAD

**Discussion by aaronhamilton** [MscrmTools/XrmToolBox#1025] [Q&A]
Two questions:
1. I have Document Locations in CRM which have lost their association to SharePointSites. Is there an XRM function I could use to re-associate? I see a method called **_Add-CrmRecordAssociation_**..... could this work? I will be attempting this next.

2. Is there a way to programmatically 'Validate' my SharePointSite records? I have tried setting a few validation fields, but it was insufficient to properly validate (ie did no fix errors on CRM's 'Files' tab).

BACKGROUND
After importing millions of client records into D365 and using XRM functions to create SharePoint Document Locations, we have encountered a problem...... 

Along with the document locations, there were corresponding SharePointSite records created, but they were flagged as "Not Validated" and therefore we were unable to navigate to SharePoint from D365's Files tab. When I manually clicked the 'Validate' button in SharePointSites entity, when we could click Files tab and see the documents in SharePoint.

So... I tried to write a PS script to set all SharePointSite records to 'Validated'... but it turned out to be insufficient to fix the Files tab, so I ran my script again to switch the IsValidated back to False. Somehow, my actions disassociated my SharePointSite record from all the corresponding Document Locations! Every document location associated with those SharePointSite records lost its sitecollectionid field. This has affected 113,000 document locations.

How to fix this situation? I see two options:
1. Delete all affected document locations and SharePoint folders and re-process them. 
2. Find a way to recreate the sitecollectionid field on all document location records using XRM functions.

**Comment by MscrmTools** [comment]
Hi,
What is the link with XrmToolBox ?

**Reply by aaronhamilton** [reply]
I am using XRMToolbox to connect to Dynamics.

**Reply by MscrmTools** [reply]
Ok but what tool are you using in XrmToolBox? You should better ask in the corresponding repository.

**Reply by aaronhamilton** [reply]
I guess that would be Microsoft.Xrm.Data.Powershell .. I didn't realize it's a different repo

**Reply by MscrmTools** [reply]
So I guess you should better ask your question [here](https://github.com/AymericM78/PowerDataOps.Xtb.Plugin/issues)
